charabia/src/normalizer/greek.rs

```diff
@@ -0,0 +1,70 @@
+use std::borrow::Cow;
+
+use super::{Normalizer, NormalizerOption};
+use crate::{Script, Token};
+
+/// Normalize Greek characters by:
+/// 1. convert final sigma into ordinary sigma
+///
+pub struct GreekNormalizer;
+
+impl Normalizer for GreekNormalizer {
+    // converting  "ς" to "σ" doesn't change the characters length,
+    // so the `normalize` method is overloaded to skip the useless char_map computing.
+    fn normalize<'o>(&self, mut token: Token<'o>, _options: NormalizerOption) -> Token<'o> {
+        if token.lemma.ends_with("ς") {
+            token.lemma = Cow::Owned(token.lemma.replace("ς", "σ"))
+        }
```

Comment:
Sorry to suggest something else than I suggested before, but I think we can target and replace only the last character:
```suggestion
        if let Some(prefix) = token.lemma.strip_suffix('ς') {
            token.lemma = Cow::Owned([prefix, "σ"].concat())
        }
```

however, this implementation loses clarity, so don't hesitate to run `cargo bench` to compare the Greek tokenization, before committing the changes. 🤔

Reply:
I've ran the benchmarks. Here are the results:

initial approach f5adfbd
<details>

![image](https://user-images.githubusercontent.com/2981598/217601689-dc2aec66-d8aa-41d8-b72f-82f2ce3f0f9f.png)

</details>

1st optimalization d0701c3 - **performance has improved**
<details>

![image](https://user-images.githubusercontent.com/2981598/217601859-2c394c67-b9eb-4d61-8266-58aa27125174.png)

</details>

`.strip_prefix` approach - **segmentation performance has regressed** - does it make sense? Can a change in normalizer affect the process of segmentation? 
**tokenize/130/Greek/Ell benchmark** -2.0098% 
other tokenize benchmarks didn't report any change  
 
<details>

![image](https://user-images.githubusercontent.com/2981598/217601925-985e9b57-e718-400d-b4cd-c0581f59344e.png)

</details>

Reply:
Well, you only worked on the Normalization which is not called during the segmentation, so let's ignore it.
The first optimization is good, skipping the char_map creation and checking only the last character works well.
About the second one, only the first benchmark shows differences, but we can easily find the reason by counting the number of words containing a ς, the first one has 4 occurrences for 130bytes whereas the second one has only 2 occurrences for 364bytes.
Let's keep the `strip_prefix` implementation, it seems to have an impact depending on the frequency of word ending by ς. 🤔

Reply:
>About the second one, only the first benchmark shows differences, but we can easily find the reason by counting the number of words containing a ς, the first one has 4 occurrences for 130bytes whereas the second one has only 2 occurrences for 364bytes.
Oh yes, that makes sense.
Maybe one of benchmarks strings should be changed, so the second benchmark case is not simpler to handle for normalizer? Then it might be easier to actually compare benchmark results. What do you think?